engine/src/main/java/org/terasology/utilities/ReflectionUtil.java

```diff
@@ -61,8 +64,14 @@ public static Type getTypeParameter(Type type, int index) {
             return (Class<?>) type;
         } else if (type instanceof ParameterizedType) {
             return (Class<?>) ((ParameterizedType) type).getRawType();
+        } else if (type instanceof GenericArrayType) {
+            GenericArrayType genericArrayType = (GenericArrayType) type;
+            return Array.newInstance(getClassOfType(genericArrayType.getGenericComponentType()), 0).getClass();
+        } else if (type instanceof WildcardType) {
+            WildcardType wildcardType = (WildcardType) type;
+            return getClassOfType(wildcardType.getUpperBounds()[0]);
         }
-        return null;
+        return Object.class;
```

Comment:
This change sticks out just a tiny bit. There are a few usages including two that check for null. Most otherwise have warnings about the possibility of a null return so this would seem like a likely improvement.

Have you checked out those usages and considered whether the swap to `Object.class` might lead to some unexpected behavior? Two places would have null checks that would be bypassed with this change (if they ever do trigger - could be impossible)

Reply:
The only two null checks are in `TypeSerializationLibrary` and `CopyStrategyLibrary`. `TypeSerializationLibrary` is going to be overhauled in this PR, so that's not a problem, and returning `Object.class` instead of `null` will not cause any problems in `CopyStrategyLibrary`. Besides, the only situation when `Object.class` will be returned is when the parameter is a `TypeVariable`, which is a very rare case.

Side Note: `CopyStrategyLibrary` could do with a very similar `CopyStrategyFactory` overhaul in the future.